src/dra/5D5BC.c

```diff
@@ -1614,30 +1614,286 @@ void DrawRichterHud(void) {
+        prim->drawMode = DRAW_HIDE;
+
+        for (altPrim = &g_PrimBuf[g_PlayerHud.primIndex2]; altPrim != NULL;
+             altPrim = altPrim->next) {
+            if (altPrim->p2 != 0) {
+                continue;
+            }
+            altPrim->drawMode = 0x2004;
+            if (altPrim->p1 != 0) {
+                altPrim->p1--;
+                continue;
+            }
+            temp_a0 = rand() & 1;
+            altPrim->y0 += temp_a0;
+            altPrim->y1 += temp_a0;
+            temp_a0 = (rand() & 3) + 1;
+            altPrim->drawMode = 0x2004;
+            altPrim->y2 += temp_a0;
+            altPrim->y3 += temp_a0;
+            if (altPrim->r2 >= 3) {
+                temp_s2 = altPrim->r2 - 3;
+                func_801071CC(altPrim, temp_s2, 2);
+                func_801071CC(altPrim, temp_s2, 3);
+            }
+            if (altPrim->y2 >= 0x100) {
+                altPrim->drawMode = 0x2015;
+                if (altPrim->r0 != 0) {
+                    temp_r0 = altPrim->r0;
+                    temp_s2 = temp_r0 & 0xFF;
+                    altPrim->r0 = temp_r0 + 0xFF;
+                    func_801071CC(altPrim, temp_s2, 0);
+                    func_801071CC(altPrim, temp_s2, 1);
+                }
+                if (altPrim->y2 >= 0x180) {
+                    altPrim->drawMode = 0x2075;
+                }
+            }
+            if (altPrim->y2 >= 0x200) {
+                altPrim->drawMode = DRAW_HIDE;
+                altPrim->p2 = 1;
+            }
+        }
+    }
+
+    prim = prim->next;
+    temp_a0 = (g_PlayerHud.displayHP * 0x5B) / g_Status.hpMax;
+    prim->y0 = prim->y1 = prim->y2 - temp_a0;
+    prim = prim->next;
+    SetPrimRect(prim, g_PlayerHud.unk14 + 0xE4, 0x70, 9, 3);
+    temp_a0 = (g_PlayerHud.unk1C * 0x5B) / 100;
+    if (temp_a0 < 0) {
+        temp_a0 = 0;
+    }
+    temp_p2 = prim->p2;
+    prim->y0 = prim->y1 = prim->y2 - temp_a0;
+    prim->p2 = temp_p2 + 0xFF;
+    if (temp_p2 == 0) {
+        prim->p1++;
+        if (prim->p1 == 9) {
+            prim->p1 = 0;
+        }
+        if (D_8003C744 == 2) {
+            prim->p2 = 1;
+        } else {
+            prim->p2 = 4;
+        }
+        prim->clut = prim->p1 + 0x103;
+    }
+    temp_s2 = prim->clut;
+    prim = prim->next;
+    SetPrimRect(prim, g_PlayerHud.unk14 + 0xEC, 0x70, 9, 3);
+    temp_a0 = (g_PlayerHud.unk20 * 0x5B) / 100;
+    if (temp_a0 >= 0x5C) {
+        temp_a0 = 0x5B;
+    }
+    prim->clut = temp_s2;
+    prim->y0 = prim->y1 = prim->y2 - temp_a0;
+    prim = prim->next;
 
-INCLUDE_ASM("dra/nonmatchings/5D5BC", DrawRichterHudSubweapon);
+    prim->u0 = (g_Status.hearts / 10) * 8;
+    prim->v0 = 0x60;
+    prim->u1 = ((g_Status.hearts / 10) * 8) + 8;
+    prim->v1 = 0x60;
+    prim->u2 = (g_Status.hearts / 10) * 8;
+    prim->v2 = 0x68;
+    prim->u3 = ((g_Status.hearts / 10) * 8) + 8;
+    prim->v3 = 0x68;
+    if ((g_Player.unk0C & 0x200000) && !(g_Timer & 2)) {
```

Comment:
My guess is that this does the flashing numbers when you have enough hearts to do an item crash